### PR TITLE
Add Download hostname override config

### PR DESCRIPTION
**Pull request by aahmed-se**

### CODE REVIEW

File: heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java [hunk 1]
```diff
@@ -89,10 +96,15 @@ public Response uploadFile(
     }
 
     String uploadDir = config.getStringValue(FILE_SYSTEM_DIRECTORY);
+    String downloadHostNameOverride = config.getStringValue(DOWNLOAD_HOSTNAME_OVERRIDE);
 
-    String fileName = UUID.randomUUID() + "-" + fileDetail.getFileName();
+    if (StringUtil.isNotBlank(downloadHostNameOverride)) {
```
Comment (cckellogg) [inline heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java]: let's add a method that we can use below when forming the uri. I don't like setting a static variable here. 
private String getHostnameOrIp(Config config);
Also you can call config.getString(key, defaultValue) so we don't need the string util.

File: heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java [hunk 2]
```diff
@@ -90,9 +97,13 @@ public Response uploadFile(
 
     String uploadDir = config.getStringValue(FILE_SYSTEM_DIRECTORY);
 
-    String fileName = UUID.randomUUID() + "-" + fileDetail.getFileName();
+    if (StringUtil.isNotBlank(getDownloadHostName())) {
```
Comment (cckellogg) [inline heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java]: can we do a little refactoring and have method call getHostnameOrIp() so we are not setting a static variable like this. Then lines 119/120 becomes:
String uri = String.format("http://%s:%s/api/v1/file/download/%s", getHostnameOrIp(), getPort(), fileName);

**Comment by aahmed-se** [comment]
@cckellogg refactored to method

File: heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java [hunk 3]
```diff
@@ -143,4 +153,12 @@ private Config createConfig() {
...
 
+  private String getHostNameOrIP() {
+    // Override hostname if provided in flags
+    if (StringUtil.isNotBlank(getDownloadHostName())) {
+      hostname = getDownloadHostName();
+    }
+    return (hostname != null) ? hostname : ip.toString();
+  }
+
 }
```
Comment (cckellogg) [inline heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java]: I think we should not be setting static state here.
final String downloadHostName = getDownloadHostName();
if (StringUtil.isNotBlank(downloadHostName) {
  return downloadHostName;
}

return hostname != null ? hostname : ip;

Comment (aahmed-se) [reply heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java]: Changed it and added a check for ip being null

**Pull request merged**